### PR TITLE
[FSTORE-1269] Extend get_feature_vector in user guide

### PR DESCRIPTION
**Pull request by kennethmhc**

### CODE REVIEW

File: docs/user_guides/fs/feature_view/feature-vectors.md [hunk 1]
```diff
@@ -37,6 +37,31 @@ You can get back feature vectors from either python or java client by providing
     featureView.getFeatureVectors(Lists.newArrayList(entry1, entry2);
     ```
 
+### Required entry
+Starting from python client v3.4, you can specify different values for the primary key of the same name which exists in multiple feature groups but is not used as the join key. The table below summarises the value of `primary_keys` in different settings. Considering that you are joining 2 feature groups, namely, `left_fg` and `right_fg`, the feature groups have different primary keys, and features (`feature_*`) in each setting. Also, the 2 feature groups are [joint](https://docs.hopsworks.ai/feature-store-api/3.7/generated/api/query_api/#join) on different *join conditions* and *prefix* as `left_fg.join(right_fg, <join conditions>, prefix=<prefix>)`.
+
+For java client, and python client before v3.4, the `primary_keys` are the set of primary key of all the feature groups in the query. These key are also compatible to python client starting from v3.4.
```
Comment (vatj) [inline docs/user_guides/fs/feature_view/feature-vectors.md:43]: "These key are also compatible to python client starting from v3.4."
What do you mean?

File: docs/user_guides/fs/feature_view/feature-vectors.md [hunk 2]
```diff
@@ -37,6 +37,31 @@ You can get back feature vectors from either python or java client by providing
...
+
+| Setting | primary key of `left_fg` | primary key of `right_fg` | join conditions                            | prefix | primary_keys                                   | note                                                     |
```
Comment (vatj) [inline docs/user_guides/fs/feature_view/feature-vectors.md:45]: Are all of these cases tested for?

File: docs/user_guides/fs/feature_view/feature-vectors.md [hunk 3]
```diff
@@ -65,7 +90,27 @@ This call will raise an exception if for example for the third entry `pk1 = 5` O
 can't be found.
 That means, `get_feature_vectors` will never return partial feature vector, but will omit empty feature vectors.
 
-If you are aware of missing featurs, you can use the [*passed features*](#passed-features) functionality, described down below.
+If you are aware of missing features, you can use the [*passed features*](#passed-features) or [Partial feature retrieval](#partial-feature-retrieval) functionality, described down below.
+
+### Partial feature retrieval
```
Comment (vatj) [inline docs/user_guides/fs/feature_view/feature-vectors.md:95]: Maybe we can point to the fraud_online tutorial to highlight a relevant use case?

**Review by vatj — CHANGES_REQUESTED** [review]

File: docs/user_guides/fs/feature_view/feature-vectors.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ Once you have trained a model, it is time to deploy it. You can get back all the
 If you want to understand more about the concept of feature vectors, you can refer to [here](../../../concepts/fs/feature_view/online_api.md).
 
 ## Retrieval
-You can get back feature vectors from either python or java client by providing the primary key value(s) for the feature view. Note that filters defined in feature view and training data will not be applied when feature vectors are returned.
+You can get back feature vectors from either python or java client by providing the primary key value(s) for the feature view. Note that filters defined in feature view and training data will not be applied when feature vectors are returned. If you need to retrieve a complete value of feature vectors without missing values, the required `entry` are [feature_view.primary_keys](https://docs.hopsworks.ai/feature-store-api/3.7/generated/api/feature_view_api/#primary_keys). Alternative, you can provide the primary key of the feature groups as the key of the entry. It is also possible to provide a subset of the entry, which will be discussed [below](#partial-feature-retrieval).
 
 === "Python"
```
[the remark this answers is not in the document]
Comment (kennethmhc) [reply docs/user_guides/fs/feature_view/feature-vectors.md:9]: This should work as expected here https://docs.hopsworks.ai/3.7/user_guides/fs/feature_view/feature-vectors/#retrieval

File: docs/user_guides/fs/feature_view/feature-vectors.md (hunk 2 above)
Comment (kennethmhc) [reply docs/user_guides/fs/feature_view/feature-vectors.md:45]: yea there are unit tests for each case: https://github.com/logicalclocks/hopsworks-ee/blob/master/hopsworks-common/src/test/io/hops/hopsworks/common/featurestore/featureview/FeatureViewControllerTest.java#L171-L709

File: docs/user_guides/fs/feature_view/feature-vectors.md (hunk 3 above)
Comment (kennethmhc) [reply docs/user_guides/fs/feature_view/feature-vectors.md:95]: We don't have an example about imputation yet. The fraud_online tutorial is a use case of passed features.

File: docs/user_guides/fs/feature_view/feature-vectors.md (hunk 1 above)
Comment (kennethmhc) [reply docs/user_guides/fs/feature_view/feature-vectors.md]: I rewrite it.

**Review by vatj — APPROVED** [review]

**Pull request merged**